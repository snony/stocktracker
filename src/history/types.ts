export interface HistoryData {
  readonly [key: string]: string | number
  readonly date: string
  readonly price: number
}

export interface HistoryState {
  history: HistoryData[]
  dateFilter: string
  priceFilter: string
}

export interface HistoryChartProps {
  readonly history: HistoryData[]
}

export interface FilterButtonProps {
  readonly type: string
  readonly value: string
  readonly selected: boolean
  readonly onClick: (symbol: string, value: string) => void
}
