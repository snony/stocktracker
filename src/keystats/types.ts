import { extractKeyStatsData } from '../api'

export type KeyStats = ReturnType<typeof extractKeyStatsData>
