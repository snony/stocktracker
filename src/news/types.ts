export enum FetchStatus {
  PENDING,
  SUCCESS,
  FAIL
}
export interface News {
  readonly url: string
  readonly headline: string
  readonly datetime: string
  readonly source: string
}

export interface NewsItemsProps {
  readonly fetchStatus: FetchStatus
  readonly newsItems: News[]
}
export interface NewsProps {
  readonly newsData: News
}
