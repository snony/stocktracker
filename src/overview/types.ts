export interface OverviewState {
  companyName: string
  symbol: string
  website: string
  description: string
}
