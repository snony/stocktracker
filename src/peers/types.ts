import { PeersState } from './peersReducer'

type Peer = string
export type Peers = Peer[]
export { PeersState }
