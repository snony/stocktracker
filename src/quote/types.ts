import { QUOTE } from 'socket/'
export interface QuoteProps {
    quote: QUOTE
    unsubscribe: () => void
}