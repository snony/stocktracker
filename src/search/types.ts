export type Company = {name: string, symbol:string};
export type ClickResult = (company:Company) => void;

export type Companies = Array<Company>

export interface SearchProps{
    readonly getInfo: ClickResult;
    readonly companySymbols: Companies; 
}

export interface SearchState{
    readonly searchValue:string;
    readonly suggestedCompanies: Companies;
}