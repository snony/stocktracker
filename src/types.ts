import api from './api'
import rootReducer from './reducer'
export type API = typeof api
export type GlobalState = ReturnType<typeof rootReducer>