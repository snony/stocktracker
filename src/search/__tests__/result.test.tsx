import { configure, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import React from 'react'
import renderer from 'react-test-renderer'
import { Company } from '../../types'
import SearchResults, { Result, ResultProps, SearchResultProps } from '../result'

function getDefaultProps(): SearchResultProps {
    return {
        results: [],
        onClickResult: () => { }
    }
}

describe('Search Results Component', () => {

    it('renders correctly with empty result passed to it', () => {

        const tree = renderer
            .create(<SearchResults {...getDefaultProps()} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('renders correctly with Results passed into it', () => {
        const results: Company[] = [{ name: 'Apple Inc', symbol: 'aapl' }, { name: 'Alphabet Inc', symbol: 'goog' }]
        const onClickResult = jest.fn()
        const resultsProp: SearchResultProps = { results, onClickResult }
        const tree = renderer
            .create(<SearchResults {...resultsProp} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })

})


describe('Result Component', () => {
    it('renders correctly ', () => {
        const onClickResult = jest.fn()
        const result: ResultProps = { company: { name: 'Apple Inc', symbol: 'aapl' }, onClickResult }
        const tree = renderer
            .create(<Result {...result} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('should be able to click company', () => {
        configure({ adapter: new Adapter() });
        const onClickResult = jest.fn()
        const result: ResultProps = { company: { name: 'Apple Inc', symbol: 'aapl' }, onClickResult }
        const element = mount(<Result {...result} />)
        element.find('li').simulate('click')
        expect(onClickResult).toHaveBeenCalledWith(result.company)
    })



})