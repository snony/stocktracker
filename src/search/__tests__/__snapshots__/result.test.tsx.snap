// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result Component renders correctly  1`] = `
.emotion-0 {
  outline: none;
  padding: 1.5rem 1rem;
  color: white;
  -webkit-transition: 0.2s ease;
  transition: 0.2s ease;
  font-size: 1rem;
}

<li
  className="emotion-0"
  onClick={[Function]}
>
  Apple Inc
   (
  aapl
  )
</li>
`;

exports[`Search Results Component renders correctly with Results passed into it 1`] = `
.emotion-2 {
  list-style-type: none;
  padding-left: 0;
  z-index: 1;
  position: absolute;
  margin: 0.7rem 0;
  width: calc(100% - 2 * 1.5rem);
}

.emotion-0 {
  outline: none;
  padding: 1.5rem 1rem;
  color: white;
  -webkit-transition: 0.2s ease;
  transition: 0.2s ease;
  font-size: 1rem;
}

<ul
  className="emotion-2"
>
  <li
    className="emotion-0"
    onClick={[Function]}
  >
    Apple Inc
     (
    aapl
    )
  </li>
  <li
    className="emotion-0"
    onClick={[Function]}
  >
    Alphabet Inc
     (
    googl
    )
  </li>
</ul>
`;

exports[`Search Results Component renders correctly with empty result passed to it 1`] = `
.emotion-0 {
  list-style-type: none;
  padding-left: 0;
  z-index: 1;
  position: absolute;
  margin: 0.7rem 0;
  width: calc(100% - 2 * 1.5rem);
}

<ul
  className="emotion-0"
/>
`;
